toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -217,6 +217,65 @@ func SetupLoopbackDevice(diskFilePath string) (devicePath string, err error) {
 	return
 }
 
+// BlockOnDiskIO waits until all outstanding operations against a disk complete.
+func BlockOnDiskIO(diskDevPath string) (err error) {
+
+	logger.Log.Infof("Flushing all IO to disk for %s", diskDevPath)
+	shell.Execute("sync")
+
+	stdout, stderr, err := shell.Execute("lsblk", "--output", "MAJ:MIN", "--noheadings", "--nodeps", diskDevPath)
+	if err != nil {
+		logger.Log.Errorf("Failed to get disk major/minor ID: %s, %s", stderr, err.Error())
+		return
+	}
+	// lsblk includes whitespace both before an after output, trim it off.
+	diskIDs := strings.Split(strings.TrimSpace(stdout), ":")
+	if len(diskIDs) != 2 {
+		return fmt.Errorf("couldn't find disk IDs for %s (%s)", diskDevPath, stdout)
+	}
+	maj := diskIDs[0]
+	min := diskIDs[1]
```

Comment:
Recently, for indeces for data retrieved from command output, we use well-named constants explaining what each index should contain. With `iota` you can usually also have an additional one at the end for the `len(X) != expectedOutputCount` check. Example: look for `listMatchSubString = iota` inside `rpmrepocloner.go`.

Reply:
I do define constants a few lines down... ah well. 

As for iota, unfortunately it won't work nicely here since they are not contiguous. I only need 0,1, and 11.

---
In reply to: [550308845](https://github.com/microsoft/CBL-Mariner/pull/481#discussion_r550308845) [](ancestors = 550308845)